Mm 44440

neallred opened the pull request:
#### Summary
Fix billing subscription page for free cloud starter, while still keeping existing experience for paid cloud starter.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-44440

#### Screenshots
![cloud-starter](https://user-images.githubusercontent.com/13738432/171286563-c99fa198-5424-4062-85fb-dffc942e78db.png)

#### Release Note
Differences in UI are behind feature flag
```release-note
NONE
```

marianunez commented:
> Mattermost test server with CWS created! 🎉
> 
> Access here: https://mattermost-webapp-pr-10431.test.mattermost.cloud
> 
> Account Type	Username	Password
> Admin	[user-mattermost-webapp-pr-10431@example.mattermost.com](mailto:user-mattermost-webapp-pr-10431@example.mattermost.com)	Cws@10102020

@neallred do we need to do a change in something to legacy Stripe to be able to see the correct Starter name for Freemium? I am seeing `Mattermost Cloud` in the test server for Freemium

marianunez commented on components/admin_console/billing/billing_summary/upsell_card.tsx:
```diff
@@ -0,0 +1,105 @@
+            id: t('admin.billing.subscriptions.billing_summary.try_enterprise'),
+            defaultMessage: 'Try Enterprise features for free',
+        }}
+        cta={{
+            id: t('admin.billing.subscriptions.billing_summary.try_enterprise.cta'),
+            defaultMessage: 'Try free for {trialLength} days',
+            values: {
+                trialLength: 30,
+            },
+        }}
```
Do we need to check here if they already had a free trial? I believe in that case then the CTA should be `Upgrade`:

https://www.figma.com/file/Aob5ffHtfjKzQoebR93ycA/Cloud-Freemium?node-id=657%3A433143

marianunez commented on components/admin_console/billing/billing_summary/upsell_card.tsx:
```diff
@@ -0,0 +1,105 @@
+    <UpsellCard
+        title={{
+            id: t('admin.billing.subscriptions.billing_summary.try_enterprise'),
+            defaultMessage: 'Try Enterprise features for free',
```
Also, this text would be "Upgrade to the Professional Plan" if they already did the trial based on Figma below

marianunez reviewed (changes requested):
Great work @neallred! Some nice refactoring included in there 💯 

Left some comments on some considerations needed for Professional and if they already started the trial before

neallred commented:
> @neallred do we need to do a change in something to legacy Stripe to be able to see the correct Starter name for Freemium? I am seeing Mattermost Cloud in the test server for Freemium

I was running on enterprise mock data which has `subscription.is_paid_tier` hard coded to true. But the CWS code doesn't set that to true unless the flag is turned off. On top of that, we have a number of confused usages of `is_paid_tier` in webapp. Opted to remove usage of `is_paid_tier` and replace it with hard coded checks for legacy free product ids. We want to retire the field anyways, and we're already doing some stuff wrong because we rely on it, and we aren't making any more legacy free plans, so once those are migrated those hard coded checks can go away.

neallred commented:
@stevemudie @marianunez this should be ready for review now. The legacy cloud text was being displayed because there was a faulty reliance on the `is_paid_tier` flag. That flag doesn't have a clear meaning for free cloud starter plans, and there was some conflicting usages of it in the code, so I opted to remove its usage altogether and instead do hard coded product id checks for legacy free plan.

I've also addressed all the PR feedback so far (different call to actions according to whether user is pre trial, post trial, or cloud professional; updated cloud professional feature listing).

marianunez commented:
@neallred in the bug bash we noticed that this subscription page is not showing well also for Enterprise and Legacy plans. Should we include them in this PR or do you prefer another ticket for it?

neallred commented:
@marianunez as its own follow up PR. I'm guessing the adjustments needed required will be more than just a couple small tweaks.

marianunez commented on components/admin_console/billing/billing_subscriptions/index.tsx:
```diff
@@ -48,14 +53,14 @@ const BillingSubscriptions: React.FC = () => {
     const contactSalesLink = useSelector((state: GlobalState) => getCloudContactUsLink(state, InquiryType.Sales));
-    const cancelAccountLink = useSelector((state: GlobalState) => getCloudContactUsLink(state, InquiryType.Sales, InquiryIssue.CancelAccount));
-    const trialQuestionsLink = useSelector((state: GlobalState) => getCloudContactUsLink(state, InquiryType.Sales, InquiryIssue.TrialQuestions));
+    const cancelAccountLink = useSelector((state: GlobalState) => getCloudContactUsLink(state, InquiryType.Sales, SalesInquiryIssue.CancelAccount));
+    const trialQuestionsLink = useSelector((state: GlobalState) => getCloudContactUsLink(state, InquiryType.Sales, SalesInquiryIssue.TrialQuestions));
```
Thanks for this additional fix 👍

marianunez reviewed (approved):
Excellent work and great refactoring/cleanup effort 🚀

pvev commented on components/admin_console/billing/billing_subscriptions/cancel_subscription.tsx:
```diff
@@ -9,17 +9,17 @@ import {trackEvent} from 'actions/telemetry_actions';
     } = props;
 
-    if (isFreeTrial || !isPaidTier) {
+    if (isFreeTrial || !isLegacyFree) {
```
nice name pick!

pvev commented on components/admin_console/billing/billing_summary/upsell_card.tsx:
```diff
@@ -0,0 +1,183 @@
+import {Message, t} from 'utils/i18n';
+import {openExternalPricingLink} from 'utils/cloud_utils';
+
+import './upsell_card.scss';
```
nit: this import should be the last one (just for consistency)

pvev commented on components/admin_console/billing/billing_summary/upsell_card.tsx:
```diff
@@ -0,0 +1,183 @@
+            id: t('admin.billing.subscriptions.billing_summary.try_enterprise.cta'),
+            defaultMessage: 'Try free for {trialLength} days',
+            values: {
+                trialLength: 30,
```
nit: good practice to keep these kind of values in constants.

pvev commented on components/admin_console/billing/plan_details/feature_list.tsx:
```diff
@@ -0,0 +1,247 @@
+    ];
+
+    const featuresCloudEnterprise = [
+        intl.formatMessage({
```
nice refactor!

pvev commented on components/admin_console/billing/plan_details/index.tsx:
```diff
@@ -5,55 +5,49 @@ import React from 'react';
 }
-const PlanDetails = ({isFreeTrial, subscriptionPlan}: PlanDetailsProps) => {
-    const locale = useSelector((state: GlobalState) => getCurrentLocale(state));
+const PlanDetails = ({isFreeTrial, subscriptionPlan}: Props) => {
```
nit: good practice to add return value `JSX.Element | null`

pvev replied:
it was not there before ;) ..

pvev commented on components/widgets/menu/menu_items/menu_cloud_trial.tsx:
```diff
@@ -122,11 +135,9 @@ const MenuCloudTrial = ({id}: Props) => {
             <a
                 className='open-see-plans-modal style-link'
-                onClick={() => null}
+                onClick={openPricingModal}
```
this was already added: https://github.com/mattermost/mattermost-webapp/pull/10440/files#diff-67522593815da10d72efdaa9a1c9700c836c36833bf6cf4b0b8f950177e25634R129

pvev commented on packages/mattermost-redux/src/selectors/entities/cloud.ts:
```diff
@@ -35,6 +40,15 @@ export function getSubscriptionProduct(state: GlobalState): Product | undefined
+    return Boolean(LegacyFreeProductIds[getCloudSubscription(state)?.product_id || '']);
+}
+
+export function checkHadPriorTrial(state: GlobalState): boolean {
```
this is nice, will use it to replace other places where the same validation is made without the selector. Thanks!

pvev reviewed (approved):
Great work @neallred ! thanks! :shipit:
